pilot/cmd/sidecar-injector/main.go

```diff
@@ -21,74 +21,59 @@ import (
 	"istio.io/istio/pilot/platform/kube/inject"
 	"istio.io/istio/pilot/tools/version"
 	"istio.io/istio/pkg/log"
 )
 
 var (
 	flags = struct {
-		kubeconfig     string
-		meshconfig     string
-		injectConfig   string
-		namespace      string
 		loggingOptions *log.Options
+
+		meshconfig       string
+		injectConfigFile string
+		certFile         string
+		privateKeyFile   string
+		port             int
 	}{
 		loggingOptions: log.NewOptions(),
 	}
 
 	rootCmd = &cobra.Command{
-		Use:   "sidecar-initializer",
-		Short: "Kubernetes initializer for Istio sidecar",
+		Use:   "sidecar-injector",
+		Short: "Kubernetes webhook for automatic Istio sidecar injection",
 		RunE: func(*cobra.Command, []string) error {
 			if err := log.Configure(flags.loggingOptions); err != nil {
 				return err
 			}
-			restConfig, client, err := kube.CreateInterface(flags.kubeconfig)
-			if err != nil {
-				return multierror.Prefix(err, "failed to connect to Kubernetes API.")
-			}
 
 			log.Infof("version %s", version.Line())
 
-			config, err := inject.GetInitializerConfig(client, flags.namespace, flags.injectConfig)
-			if err != nil {
-				return multierror.Prefix(err, "failed to read initializer configuration")
-			}
-
-			// retrieve mesh configuration separately
-			if config.Params.Mesh, err = cmd.ReadMeshConfig(flags.meshconfig); err != nil {
-				return multierror.Prefix(err, "failed to read mesh configuration.")
-			}
-
-			initializer, err := inject.NewInitializer(restConfig, config, client)
+			wh, err := inject.NewWebhook(flags.injectConfigFile, flags.meshconfig, flags.certFile, flags.privateKeyFile, flags.port) // nolint: lll
 			if err != nil {
-				return multierror.Prefix(err, "failed to create initializer")
+				return multierror.Prefix(err, "failed to create injection webhook")
 			}
 
 			stop := make(chan struct{})
-
-			go initializer.Run(stop)
-
+			go wh.Run(stop)
 			cmd.WaitSignal(stop)
 			return nil
 		},
 	}
 )
 
 func init() {
-	rootCmd.PersistentFlags().StringVar(&flags.kubeconfig, "kubeconfig", "",
-		"Use a Kubernetes configuration file instead of in-cluster configuration")
-	rootCmd.PersistentFlags().StringVar(&flags.meshconfig, "meshconfig", "/etc/istio/config/mesh",
+	rootCmd.PersistentFlags().StringVar(&flags.meshconfig, "meshConfig", "/etc/istio/config/mesh",
 		"File name for Istio mesh configuration")
-	rootCmd.PersistentFlags().StringVar(&flags.injectConfig, "injectConfig", "istio-inject",
-		"Name of initializer configuration ConfigMap")
-	rootCmd.PersistentFlags().StringVar(&flags.namespace, "namespace", v1.NamespaceDefault, // TODO istio-system?
-		"Namespace of initializer configuration ConfigMap")
+	rootCmd.PersistentFlags().StringVar(&flags.injectConfigFile, "injectConfig", "/etc/istio/inject/config",
+		"File name for Istio injection configuration")
+	rootCmd.PersistentFlags().StringVar(&flags.certFile, "tlsCertFile", "/etc/istio/certs/cert.pem",
+		"File containing the default x509 Certificate for HTTPS.")
```

Comment:
Do we ship a default cert or do we mean the default location?

Reply:
Just the location. I'll change the wording to make it less confusing.